src/client/Microsoft.Identity.Client/MsalErrorMessage.cs

```diff
@@ -285,6 +285,7 @@ public static string RedirectUriMismatch(string expectedUri, string actualUri)
             "You can modify the configuration in the application registration portal. See https://aka.ms/msal-net-invalid-client for details. ";
         public const string SSHCertUsedAsHttpHeader = "MSAL was configured to request SSH certificates from AAD, and these cannot be used as an HTTP authentication header. Developers are responsible for transporting the SSH certificates to the target machines.";
         public const string BrokerApplicationRequired = "Installation of broker failed. The broker application must be installed to continue authentication";
+        public const string NoUPNForSilentAuth = "Unable to query the android account manager because no username is provided in the silent authentication request. Please try using AcquireTokenSilent(IEnumerable<string> scopes, string loginHint). See https://aka.ms/Brokered-Authentication-for-Android for details";
```

Comment:
nit: suggestion: `NoUpnForSilentBrokerAuth`

And, this can be because no username or login hint was passed correct? like both were null?

Reply:
Yes. I modified this a little and now it is used in a msaluirequired exception instead of a client exception

Reply:
for this error message, the only time they'd get this is if using their own IAccount object, which might not have all the stuff in it...can you add an aka.ms link or have something more actionable? like what should the developer do to mitigate this or how did this happen in the first place? thx.

---
In reply to: [382296441](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1649#discussion_r382296441) [](ancestors = 382296441)

Reply:
+1: to the aka.ms link!

---
In reply to: [382302642](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1649#discussion_r382302642) [](ancestors = 382302642,382296441)